File: namoosori_stage3/namoosori-contents-source/phase1/javastory.club.stage3/src/javastory/club/stage3/step4/ui/menu/ClubMenu.java
package javastory.club.stage3.step4.ui.menu;

import javastory.club.stage3.step4.ui.console.ClubConsole;
import javastory.club.stage3.util.Narrator;
import javastory.club.stage3.util.TalkingAt;

import java.util.Scanner;

public class ClubMenu {

    private ClubMembershipMenu clubMembershipMenu;
    private ClubConsole clubConsole;

    private Scanner sc;
    private Narrator narrator;

    public ClubMenu() {

        this.clubMembershipMenu = new ClubMembershipMenu();
        this.clubConsole = new ClubConsole();
        this.sc = new Scanner(System.in);
        this.narrator = new Narrator(this, TalkingAt.Left);
    }

    public void show(){
        int inputNum = 0;
        while(true){
            displayMenu();
            inputNum = selectMenu();

            switch (inputNum) {

                case 1:
                    clubConsole.register();
                    break;
                case 2:
                    clubConsole.find();
                    break;
                case 3:
                    clubConsole.modify();
                    break;
                case 4:
                    clubConsole.remove();
                    break;
                case 5:
                    clubMembershipMenu.show();
                    break;
                case 0:
                    return;

                default:
                    narrator.sayln("Choose again!");
            }
        }
    }

    private void displayMenu() {
        narrator.sayln("");
        narrator.sayln("..............................");
        narrator.sayln(" Club menu ");
        narrator.sayln("..............................");
        narrator.sayln(" 1. Register");
        narrator.sayln(" 2. Find");
        narrator.sayln(" 3. Modify");
        narrator.sayln(" 4. Remove");
        narrator.sayln("..............................");
        narrator.sayln(" 5. Membership menu");
        narrator.sayln("..............................");
        narrator.sayln(" 0. Previous");
        narrator.sayln("..............................");
    }

    private int selectMenu() {
        System.out.print("Select: ");
        int menuNumber = sc.nextInt();

        if (menuNumber >= 0 && menuNumber <= 5) {
            sc.nextLine();
            return menuNumber;
        } else {
            narrator.sayln("It's a invalid number --> " + menuNumber);
            return -1;
        }
    }
}

File: nextree/React+MobX+TypeScript/travelclubtest/src/menu/ClubMenu.ts
import { question } from 'readline-sync';
import ClubWindow from '../console/ClubConsole';

class ClubMenu {
    
    private _clubWindow : ClubWindow;

    constructor(){
        this._clubWindow = new ClubWindow();
    }

    displayMenu() {
        console.log("");
        console.log("..............................");
        console.log(" Club menu ");
        console.log("..............................");
        console.log(" 1. Register");
        console.log(" 2. Find");
        console.log(" 3. Modify");
        console.log(" 4. Remove");
        console.log("..............................");
        console.log(" 5. Membership menu");
        console.log("..............................");
        console.log(" 0. Previous"); //임시로 exit 으로 설정
        console.log("..............................");
    }

    selectMenu() {
        let menuNumber = parseInt(question('select : '));
        if (menuNumber >= 0 && menuNumber <= 5) {
            return menuNumber;
        } else {
            console.log('Invalid number => ' + menuNumber);
            return -1;
        }
    }

    exitProgram() {
        console.log('종료합니다..');
        process.exit(0);
    }

    show() {
        let inputNumber = 0;

        while (true) {
            this.displayMenu();
            inputNumber = this.selectMenu();
            switch (inputNumber) {
                case 1: 
                this._clubWindow.register();
                break;
                case 2:
                this._clubWindow.find();    
                break;
                case 3: 
                this._clubWindow.modify();
                break;
                case 4:
                this._clubWindow.remove();    
                break;
                case 5: break;
                case 0: this.exitProgram(); break;

            }
        }
    }
}

new ClubMenu().show();